lib/open_feature/client.rb

```diff
@@ -218,6 +235,29 @@ def fetch_structure_details(flag_key:, default_value:, context: nil, options: ni
     sig { returns(Provider) }
     attr_reader :provider
 
+    sig do
+      type_parameters(:U).params(
+        flag_key: String,
+        default_value: T.type_parameter(:U),
+        invocation_context: T.nilable(EvaluationContext),
+        options: T.nilable(EvaluationOptions),
+        flag_type: String
+      ).returns(EvaluationContext)
+    end
+    def build_context_with_before_hooks(flag_key:, default_value:, invocation_context:, options:, flag_type:)
+      evaluation_context = build_context(invocation_context) || OpenFeature::EvaluationContext.new
+      hook_context = HookContext.new(flag_key: flag_key, default_value: default_value,
+                                     evaluation_context: evaluation_context, flag_type: flag_type,
+                                     client_metadata: client_metadata, provider_metadata: provider.metadata)
+      OpenFeature::Hook::BeforeHook.call(hooks: sequence_before_hooks(options), context: hook_context,
+                                         hints: {})
+    end
+
+    sig { params(options: T.nilable(EvaluationOptions)).returns(T::Array[Hook]) }
+    def sequence_before_hooks(options)
```

Comment:
Ditto about moving this out to it's own class/method on another module. I'd love to test this in isolation.

Reply:
I put this in the `Hooks` value class. lmk your thoughts